### PR TITLE
Add Arabic courses

### PR DESCRIPTION
**Pull request by AhmedKhairyM0**
## What does this PR do?
Add resource(s)

### Why is this valuable (or not)?
I added courses on OOP paradigm, SOLID principles, operating systems, logic programming, and Firebase in Flutter.

### How do we know it's really free?
All my resources are from YouTube.
 
## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [ ] Search for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by davorpa** [comment]
:information_source: It's usually not helpful to add a category for a single resource. :twisted_rightwards_arrows: Try to...
1. ... take a look at other files to know how to are organized
2. ... find another category/section matching their topics.
3. ... see if there are other resources of same genre (books, courses...) in our lists or third-party that would fit into this newer. This fact could contribute to consolitate it.

**Review by davorpa — COMMENTED** [review]
:information_source: It's usually not helpful to add a category for a single resource. :twisted_rightwards_arrows: Try to...
1. ... take a look at other files to know how to are organized
2. ... find another category/section matching their topics.
3. ... see if there are other resources of same genre (books, courses...) in our lists or third-party that would fit into this newer. This fact could help to consolidate it.

**Comment by AhmedKhairyM0** [comment]
I tried to do your guidelines but there are some topics like operating systems, I can't categorize them
 @davorpa

**Comment by AhmedKhairyM0** [comment]
I tried to do your guidelines but there are some topics such as operating systems. they can't be categorized.
@davorpa

**Comment by AhmedKhairyM0** [comment]
For "Logic Programming - Prolog" should I change its name to "Prolog" or "Logic Programming" or leave it?
@davorpa

File: courses/free-courses-ar.md [hunk 1]
```diff
@@ -22,7 +25,10 @@
   * [PugJs](#pugjs)
   * [React.js](#reactjs)
   * [Vue.js](#vuejs)
+* [Logic Programming - Prolog](#logic-programming---prolog)
```
Comment (davorpa) [inline courses/free-courses-ar.md:28]: In english list by lang each one has it own category: `Prolog`, `Haskell`, `Lisp`...

File: courses/free-courses-ar.md [hunk 2]
```diff
@@ -214,6 +194,21 @@
...
 
 
+### Logic Programming
+
+* [Logic Programming - Prolog - برمجة منطقية](https://www.youtube.com/playlist?list=PLMm8EjqH1EFW9Faldu6D6Uh2j1EWWaTYe) - Ahmed Sallam
+
+
+### Object Oriented Programming
+
+* [Object-Oriented Programming JAVA in Arabic](https://www.youtube.com/playlist?list=PLCInYL3l2AagY7fFlhCrjpLiIFybW3yQv) - Adel Nasim
```
Comment (davorpa) [inline courses/free-courses-ar.md]: This matches at JAVA category

**Comment by eshellman** [comment]
It's usually not helpful to add a category for a single resource. Remember, the goal is to help users find books of interest, not to create a global ontology. That means multiple items should be grouped together whenever possible.

File: courses/free-courses-ar.md [hunk 3]
```diff
@@ -7,6 +7,8 @@
 * [C](#c)
 * [C Sharp](#c-sharp)
 * [Clean Code](#clean-code)
```
Comment (davorpa) [inline courses/free-courses-ar.md:9]: Seen [subject lists](https://github.com/EbookFoundation/free-programming-books/blob/master/books/free-programming-books-subjects.md#software-architecture) I think is better join all this branch (your additions included) into same category and renaming to `Software Architecture`

File: courses/free-courses-ar.md [hunk 4]
```diff
@@ -257,6 +238,34 @@ Emam Academy
...
 
 
+### Logic Programming - Prolog
+
+* [Logic Programming - Prolog - برمجة منطقية](https://www.youtube.com/playlist?list=PLMm8EjqH1EFW9Faldu6D6Uh2j1EWWaTYe) - Ahmed Sallam
+
+
+### Object Oriented Paradigm
+
+* [C++ Object-Oriented Design and Programming](https://www.youtube.com/playlist?list=PLPt2dINI2MIbMba7tpx3qvmgOsDlpITwG) -  Arabic Competitive Programming
```
Comment (davorpa) [inline courses/free-courses-ar.md]: Better in C++ category

File: courses/free-courses-ar.md [hunk 5]
```diff
@@ -22,7 +25,10 @@
   * [PugJs](#pugjs)
   * [React.js](#reactjs)
   * [Vue.js](#vuejs)
+* [Logic Programming - Prolog](#logic-programming---prolog)
 * [.NET](#net)
+* [Object Oriented paradigm](#object-oriented-paradigm)
```
Comment (davorpa) [inline courses/free-courses-ar.md:30]: This was pending to delete between so much changes :wink: 

Remove patch:
```suggestion
```

File: courses/free-courses-ar.md [hunk 6]
```diff
@@ -120,7 +143,10 @@
 
 ### Flutter
 
+* [Advanced Flutter Tutorial - build E commerce App](https://www.youtube.com/playlist?list=PLGVaNq6mHiniedDoXJd35XFBNvJAoq-xe) - Scholar Tech (:construction: *in process*)
 * [Flutter بالعربى](https://www.youtube.com/playlist?list=PLMDrOnfT8EAhsiJwkzspHp_Ob6oRCHxv0) - Asem Saafan
+* [Flutter BLoC - بالعربي](https://www.youtube.com/playlist?list=PLwWuxCLlF_ufA0GYYjlx_R4smekKH_AuB) - Omar Ahmed
+* [Mastering Firebase](https://www.youtube.com/playlist?list=PL93xoMrxRJIve-GSKU61X6okh5pncG0sH) - Wael Abu hamza
```
Comment (davorpa) [inline courses/free-courses-ar.md:149]:
```suggestion
* [Mastering Firebase And Flutter version 2 (2021) - Course - project -فلاتر- مشروع - دورة - فايربيز - شرح - احتراف - كورس](https://www.youtube.com/playlist?list=PL93xoMrxRJIve-GSKU61X6okh5pncG0sH) - Wael Abu hamza
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: courses/free-courses-ar.md [hunk 7]
```diff
@@ -2,10 +2,13 @@
 
 * [Algorithms & Data Structures](#algorithms--data-structures)
 * [Assembly](#assembly)
+  * [Computer Architecture](#computer-architecture)
```
Comment (davorpa) [inline courses/free-courses-ar.md:5]: This better as 3th level (as same as it was at first commits), like in other listings

File: courses/free-courses-ar.md (hunk 2 above)
Comment (AhmedKhairyM0) [reply courses/free-courses-ar.md]: Yes, when I noticed it was there I erased it.

File: courses/free-courses-ar.md (hunk 4 above)
Comment (AhmedKhairyM0) [reply courses/free-courses-ar.md]: Yes, it's best in C++, so when I noticed it was there I erased it.

File: courses/free-courses-ar.md (hunk 3 above)
Comment (AhmedKhairyM0) [reply courses/free-courses-ar.md:9]: Great !
But do the repo rules allow me to use a 3rd level header as a subtitle?
As in this picture 
Screenshot from 2021-11-01 00-40-00

Comment (AhmedKhairyM0) [reply courses/free-courses-ar.md:9]: Did you mean to rename "Clean Code"  to "Software Architecture"?

Comment (AhmedKhairyM0) [reply courses/free-courses-ar.md:9]: Do you want me to write them as
- Software Architecture
  - Clean Code
  - Design Patterns
  - Solid

Comment (davorpa) [reply courses/free-courses-ar.md:9]: Plain. Only `Software Architecture`. Move all resources there

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by AhmedKhairyM0** [comment]
You're welcome, I am happy to contribute to this wonderful work.
@eshellman

**Pull request merged**